tgui/packages/tgui/interfaces/Microscope.tsx

```diff
@@ -0,0 +1,183 @@
+      <Stack key={cell_line.desc} mt={2}>
+        <Stack.Item>
+          <DmIcon
+            icon={cell_line.icon}
+            icon_state={cell_line.icon_state}
+            height="64px"
+            width="64px"
+          />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Section
+            title={cell_line.desc}
+            buttons={
+              <Button
+                color="transparent"
+                icon="circle-question"
+                tooltip="Put the sample into a Growing Vat and pour the required reagents."
+              />
+            }
+          >
+            <Box my={1}>
+              Consume {cell_line.consumption_rate} units of every nutrient per
+              second to grow by {cell_line.growth_rate}%.
+            </Box>
+            {cell_line.suspectibility > 0 && (
+              <Box my={1}>
+                Reduced by {cell_line.suspectibility}% when infected with
+                viruses.
+              </Box>
+            )}
+            <Stack fill>
+              <Stack.Item grow>
+                <GroupTitle title="Required Reagents" />
+                {Object.keys(cell_line.requireds).map((reagent) => (
+                  <Button fluid key={reagent}>
+                    {reagent}
+                  </Button>
+                ))}
+              </Stack.Item>
+              <Stack.Item grow>
+                <GroupTitle title="Supplements" />
+                {Object.keys(cell_line.supplementaries).map((reagent) => (
+                  <Button
+                    fluid
+                    color="good"
+                    key={reagent}
+                    tooltip={
+                      '+' + cell_line.supplementaries[reagent] + '% growth/sec.'
+                    }
+                  >
+                    {reagent}
+                  </Button>
+                ))}
+              </Stack.Item>
+              <Stack.Item grow>
+                <GroupTitle title="Supressives" />
+                {Object.keys(cell_line.suppressives).map((reagent) => (
+                  <Button
+                    fluid
+                    color="bad"
+                    key={reagent}
+                    tooltip={cell_line.suppressives[reagent] + '% growth/sec.'}
+                  >
+                    {reagent}
+                  </Button>
+                ))}
+              </Stack.Item>
+            </Stack>
+          </Section>
+        </Stack.Item>
+      </Stack>
+    ) : (
+      <Stack key={cell_line.desc} mt={2}>
+        <Stack.Item>
+          <Icon name="viruses" color="bad" size={4} mr={1} />
+        </Stack.Item>
+        <Stack.Item grow>
+          <Section title={cell_line.desc}>
+            <Box my={1}>
+              Reduces growth of other cell lines when not suppressed by
+              Spaceacillin.
+            </Box>
+          </Section>
+        </Stack.Item>
+      </Stack>
+    );
+  });
+};
+
+const GroupTitle = ({ title }) => {
```

Comment:
another not-required bit i'd like to add is that destructuring in the args this does not mark it as const, it's mutable, and with no actual type declaration it's also any

this is a trivial difference in this context (therefore optional) but we can preserve function signatures and make props immutable (as they should be) by not inlining like this